### PR TITLE
 haskellPackages.jsaddle-hello: fix on darwin 

### PR DESCRIPTION
**Pull request by alexfmpe**
Strictly speaking, `ghcjs-dom-hello` fails with a linker error on my mac but I suspect that's just for aarch64-darwin since I've often hit https://github.com/NixOS/nixpkgs/issues/149692. Should be fixed when the cctools bump is merged.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by maralorn** [comment]
sterni apparently wants to comment on this as well. I think this largely makes sense.

File: pkgs/development/haskell-modules/configuration-darwin.nix [hunk 1]
```diff
@@ -323,6 +323,20 @@ self: super: ({
   # Tests fail on macOS https://github.com/mrkkrp/zip/issues/112
   zip = dontCheck super.zip;
 
+  warp = super.warp.overrideAttrs (drv: {
+    __darwinAllowLocalNetworking = true;
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-darwin.nix:327]: How did you hit this? Seems like building `warp` hasn't been a problem so far.

Comment (alexfmpe) [reply pkgs/development/haskell-modules/configuration-darwin.nix:327]: Might be some combination of sandbox + firewall. I'll try to isolate the trigger

Comment (alexfmpe) [reply pkgs/development/haskell-modules/configuration-darwin.nix:327]: This builds for me with `--option sandbox false`, does not build with `relaxed` or `true`.
Test suite just hangs after the `"range requests"` group log and there's no more output:
https://github.com/yesodweb/wai/blob/d50e1184600631a114e3d2ad119abdf0ef08834b/warp/test/ResponseSpec.hs#L106
I suspect this is the first case in the suite that actually performs a network request (apparently in `127.0.0.1` flavor)

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-darwin.nix:327]: Do you hit this with just GHC 9.8? warp does seem to succeed with GHC 9.6 which is confusing to me.

Comment (alexfmpe) [reply pkgs/development/haskell-modules/configuration-darwin.nix:327]: On further testing it looks like it only happens when both firewall AND sandboxing are active. 
Didn't try 9.8.

**Pull request merged**